sdk/Sdk.Generators/ISymbolExtensions.cs

```diff
@@ -25,13 +25,22 @@ internal static string GetFullName(this ITypeSymbol typeSymbol)
             }
 
             var sb = new StringBuilder(symbol.MetadataName);
-            symbol = symbol.ContainingSymbol;
 
-            while (!IsRootNamespace(symbol))
+            if (symbol is IArrayTypeSymbol arraySymbol) // arrays need to be handled differently b/c the properties used to get the full name for other symbols are null for IArrayTypeSymbols
```

Comment:
I was looking at the usages of this function and see most of it is for comparing if a given type matches an expected type. This seems odd to me we are comparing these by string. There has to be a stronger typed way to verify type matching. This is for another PR, but I _think_ we should invert this via `compilation.GetTypeByMetadataName` then compare that to the `ITypeSymbol` we already have.

So instead of `string.Equals(typeof(Task).FullName, typeSymbol.GetFullName())` we should alternatively do `typeSymbol.Equals(compilation.GetTypeByMetadataName("System.Threading.Tasks.Task"))`.

I am only bringing this up because you are enhancing logic for a use-case which we can approach from a different (and more reliable) way.